docs/variables.md

```diff
@@ -0,0 +1,108 @@
+<!--
+---
+linkTitle: "Variables"
+weight: TODO
+---
+-->
+# Variables
+
+This doc aggregates the complete set of variable substitions available
+in `Tasks` and `Pipelines`.
+
+## Task Variables
+
+| Variable | Description |
+| -------- | ----------- |
+| `params.<param name>` | The value of the param at runtime. |
+| `resources.inputs.<resource name>.path` | The path to the input resource's directory. |
+| `resources.outputs.<resource name>.path` | The path to the output resource's directory. |
+| `results.<result name>.path` | The path to the file where a Task's result should be written. |
```

Comment:
I thought it was possible for a Task to refer to another task results now?

Reply:
Something like:

```
tasks.[task-name].results.[result-name]
```

Reply:
Ah, this is a Pipeline Task variable, which got pushed down under the PipelineResource variables since they're Task-only i think.  I'll move the Pipeline variables up higher.

Reply:
Oh, right, I missed that, thank you!